infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/VerifyingRemoteLeaderState.java

```diff
@@ -55,6 +55,9 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                     return fsm.getStates().get(LogReplicationRuntimeStateType.WAITING_FOR_CONNECTIVITY);
                 }
                 return this;
+            case REMOTE_LEADER_LOSS:
```

Comment:
but since in this state you don't have a remote leader, you can just ignore it, right?... It would throw an IllegalTransitionException which just ends up ignoring the event here. BTW we might want to change that logging level of the IllegalTransitionException to something lower as it is not really critical like we discussed the other day.

Reply:
The issue was during the Replicating State. We assume that we can only move to the VerifyingRemoteLeader only if the connection is down. But this is not the only scenario when it can get lost. It can also get lost if it loses leadership on the remote. In that case, it actually explicitly notifies the active about it every time actively tries to send a message.